controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -20,65 +20,98 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/go-logr/logr"
 	v1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	"github.com/red-hat-storage/ocs-operator/api/v1alpha1"
 	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
 	"github.com/red-hat-storage/ocs-operator/controllers/storagecluster"
+	controllers "github.com/red-hat-storage/ocs-operator/controllers/storageconsumer"
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const (
+	storageClassClaimFinalizer  = "storagesclassclaim.ocs.openshift.io"
+	StorageClassClaimAnnotation = "ocs.openshift.io.storagesclassclaim"
+)
+
 // StorageClassClaimReconciler reconciles a StorageClassClaim object
 // nolint
 type StorageClassClaimReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
+	log    logr.Logger
+	ctx    context.Context
+
+	storageConsumer              *ocsv1alpha1.StorageConsumer
+	storageCluster               *v1.StorageCluster
+	storageClassClaim            *ocsv1alpha1.StorageClassClaim
+	cephBlockPool                *rookCephv1.CephBlockPool
+	cephFilesystemSubVolumeGroup *rookCephv1.CephFilesystemSubVolumeGroup
+	cephClientProvisioner        *rookCephv1.CephClient
+	cephClientNode               *rookCephv1.CephClient
+	cephResourcesByName          map[string]*ocsv1alpha1.CephResourcesSpec
+
+	namespace string
 }
 
 // +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=ceph.rook.io,resources=cephclients,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=ceph.rook.io,resources=cephfilesystemsubvolumegroups,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageconsumers/status,verbs=get;update;patch
```

Comment:
Do we really want to update or patch the storage consumer status via the storage-class claim controller?

Reply:
no, earlier it was updating but not now, removed.